domains/client/domain-operator/src/domain_bundle_producer.rs

```diff
@@ -119,6 +122,7 @@ where
         >,
         bundle_sender: Arc<BundleSender<Block, CBlock>>,
         keystore: KeystorePtr,
+        skip_empty_bundle_production: bool,
```

Comment:
Not too happy about this change. Why not just make `non_empty_er_exists` returns true all the time with neccessary explanation on the test-runtime ?

Reply:
I also want to add a case to test how it works when the operator does skip producing empty bundle, we can't achieve that if the flag is hardcoded in runtime.

Reply:
I dont understand. If the `non_empty_er_exists` always returns true, then operator should always produce bundle. I'm not sure why the test is necessary to do that. Adding this boolean flag seems like a hack and looks weird. 

The other option is to put the flag under `#[cfg(test)]` so this wont go into production code.

Reply:
We want `non_empty_er_exists ` to return `false` for the test `test_skip_empty_bundle_production` and return `true` for the rest. `#[cfg(test)]` will make things more complicated since the flag is passing around many places.

No strong opinion, if you do think `test_skip_empty_bundle_production` is unnecessary I will remove it along with the flag.

Reply:
my concern here is, in our reference implementation, only client chose whether to produce a bundle or not which is already a bit of hack. Introducing this field in our reference is even more hackier solution. 

I would ideally like for the runtime to decide if the bundle production should happen but that is out of scope for this PR.

I'll leave it up to you on this since adding flag or not does not make any new difference

Reply:
The field `skip_empty_bundle_production` is not really an option that is exposed to the user or is configurable, it is just used internally for testing purposes.

In another aspect, we didn't enforce whether the operator can or not produce empty bundle, both are legitimate, no one will get slashed for a different choice. So if an operator wants, it can modify the reference implementation to produce empty bundle (though it is not economically rational).